osgar/drivers/deedee.py

```diff
@@ -0,0 +1,165 @@
+import datetime
+import io
+import math
+import struct
+
+from osgar.node import Node
+
+SLIP_END = 0xC0
+SLIP_ESCAPE = 0xDB
+SLIP_ESCAPED_END = 0xDC
+SLIP_ESCAPED_ESCAPE = 0xDD
+
+
+def slip_stream():
+    SLIP_STATE_NORMAL, SLIP_STATE_ESCAPED = list(range(2))
+    state = SLIP_STATE_NORMAL
+    buf = []
+    packets = []
+    while True:
+        raw = (yield packets)
+        packets = []
+        for b in list(raw):
+            if state == SLIP_STATE_NORMAL:
+                if b == SLIP_ESCAPE:
+                    state = SLIP_STATE_ESCAPED
+                elif b == SLIP_END:
+                    if not buf:
+                        # Ignore empty frames.
+                        continue
+                    packets.append(bytes(buf))
+                    buf = []
+                else:
+                    buf.append(b)
+            elif b == SLIP_ESCAPED_ESCAPE:
+                buf.append(SLIP_ESCAPE)
+                state = SLIP_STATE_NORMAL
+            elif b == SLIP_ESCAPED_END:
+                buf.append(SLIP_END)
+                state = SLIP_STATE_NORMAL
+            else:
+                print('SLIP: Received and discarded a corrupted message.')
+                buf = []
+                state = SLIP_STATE_NORMAL
+
+
+class Slip(Node):
+    def __init__(self, config, bus):
+        super().__init__(config, bus)
+        bus.register('packet', 'raw')
+        self.slip_stream = slip_stream()
+        next(self.slip_stream)  # Warmimg the generator up.
+
```

Comment:
I would maybe remove the double empty lines -> single empty line, but OK for ver0